src/main/java/org/currencygoldexchangeapp/services/UserMenuService.java

```diff
@@ -0,0 +1,177 @@
+package org.currencygoldexchangeapp.services;
+
+import org.currencygoldexchangeapp.datamodels.CurrencyExchange;
+import org.currencygoldexchangeapp.datamodels.GoldValue;
+import org.currencygoldexchangeapp.exceptions.CurrencyNotFoundException;
+import org.currencygoldexchangeapp.exceptions.DataNotFoundException;
+import org.currencygoldexchangeapp.handlers.ExchangeRateAPIHandler;
+import org.currencygoldexchangeapp.handlers.ExchangeRateFileReaderHandler;
+import org.currencygoldexchangeapp.handlers.ExchangeRateFileSaverHandler;
+import org.currencygoldexchangeapp.handlers.GoldValueAPIHandler;
+import org.currencygoldexchangeapp.utils.InputUtility;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.net.http.HttpClient;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class UserMenuService {
+    private final Scanner scanner = new Scanner(System.in);
+    private final ExchangeRateFileReaderHandler exchangeRateFileReaderHandler;
+    private final ExchangeRateFileSaverHandler exchangeRateFileSaverHandler;
+    private final GoldValueAPIHandler goldValueAPIHandler;
+    private final CurrencyExchangeCalculateService currencyExchangeCalculateService;
+    private final GoldValueCalculateService goldValueCalculateService;
+
+    public UserMenuService() {
+        ExchangeRateAPIHandler exchangeRateAPIHandler = new ExchangeRateAPIHandler(HttpClient.newHttpClient());
+        this.currencyExchangeCalculateService = new CurrencyExchangeCalculateService(exchangeRateAPIHandler);
+        this.exchangeRateFileReaderHandler = new ExchangeRateFileReaderHandler(currencyExchangeCalculateService);
+        this.exchangeRateFileSaverHandler = new ExchangeRateFileSaverHandler();
+        this.goldValueAPIHandler = new GoldValueAPIHandler(HttpClient.newHttpClient());
+        this.goldValueCalculateService = new GoldValueCalculateService(goldValueAPIHandler);
+    }
+
+    public void runApplication() {
+        int choice = 0;
+        while (choice != 5) {
+            System.out.println();
```

Comment:
Can you put some first time user friendly description of the application? Just the menu here is not enough IMO

Reply:
Of course, its a good idea to give a user more information about what he can achieve by using this application.